python/.gitignore

```diff
@@ -0,0 +1,4 @@
+.idea
+poetry.lock
+src/__pycache__/*
+test/__pycache__/*
```

Comment:
i would remove `.idea` and encourage everybody to use a global gitignore (otherwise this file is "open to modification" for any tool that a developer uses in their own env)